run-e2e-test.sh

```diff
@@ -0,0 +1,173 @@
+    printf '%s\n' "$OC_BIN"
+    return 0
+  fi
+
+  # 1) PATH lookup
+  local cand
+  cand="$(command -v oc 2>/dev/null || true)"
+  if [ -n "$cand" ] && [ -x "$cand" ]; then
+    printf '%s\n' "$cand"
+    return 0
+  fi
+
+  # 2) Common locations
+  for p in ./oc ./bin/oc "$HOME/.local/bin/oc" /usr/local/bin/oc /usr/bin/oc; do
+    if [ -x "$p" ]; then
+      printf '%s\n' "$p"
+      return 0
+    fi
+  done
+
+  return 1
+}
+
+# Find a usable flightctl binary location
+find_flightctl() {
+  if [ -n "${FLIGHTCTL_BIN:-}" ] && [ -x "$FLIGHTCTL_BIN" ]; then
+    printf '%s\n' "$FLIGHTCTL_BIN"
+    return 0
+  fi
+
+  # 1) PATH lookup
+  local cand
+  cand="$(command -v flightctl 2>/dev/null || true)"
+  if [ -n "$cand" ] && [ -x "$cand" ]; then
+    printf '%s\n' "$cand"
+    return 0
+  fi
+
+  # 2) Common locations
+  for p in ./flightctl ./bin/flightctl "$HOME/.local/bin/flightctl" /usr/local/bin/flightctl /usr/bin/flightctl; do
+    if [ -x "$p" ]; then
+      printf '%s\n' "$p"
+      return 0
+    fi
+  done
+
+  return 1
+}
+
+if [ $USER == "kni" ]; then
+    echo "Generate virtual agent machine"
+    login_oc
+    login_flightctl
+    #TODO: generate virtual agent machine
+    cd cypress
+    if [ ! -d "node_modules" ]; then
+     echo "Installing Cypress dependencies..."
+     npm install cypress --save-dev
+    fi
+
+    if [ $# -eq 0 ]; then
+      echo "Runs Cypress tests in headless mode"
+      echo "Usage: run-e2e-test browser true"
+      echo "  '<browser>' is the browser in which the test will be run, permitted values are chrome or firefox"
+      echo "  'true' runs Cypress in headless mode. When omitted, launches Cypress Test Runner"
+      echo "Examples:"
+      echo "  run-e2e-test.sh                                       // displays this help text"
+      echo "  run-e2e-test.sh firefox true                    // runs all e2e tests in headless mode on firefox"
+      trap EXIT
+      exit;
+    fi
+
+    if [ $# -eq 0 ]; then
+      echo "Runs Cypress tests in headless mode"
+      echo "Usage: run-e2e-test browser true"
+      echo "  '<browser>' is the browser in which the test will be run, permitted values are chrome or firefox"
+      echo "  'true' runs Cypress in headless mode. When omitted, launches Cypress Test Runner"
+      echo "Examples:"
+      echo "  run-e2e-test.sh                                       // displays this help text"
+      echo "  run-e2e-test.sh firefox true                    // runs all e2e tests in headless mode on firefox"
+      trap EXIT
+      exit;
+    fi
+
+    if [ -n "$2" ] && [ -n "$1" ]; then
+      npx cypress run --browser $1
+      exit;
+    fi
+    if [ -n "$1"] && [ "$2" == "false"]; then
+      npx cypress open
+      exit;
+    fi
+else
+  echo "Generate virtual agent machine"
+  login_oc
+  login_flightctl
+
+  cd cypress
+  if [ ! -d "node_modules" ]; then
+    echo "Installing Cypress dependencies..."
+    npm install cypress --save-dev
```

Comment:
Here, and in the `README.md` we are not pinning the cypress to any version.
Since we cannot guarantee that the tests will run in newer versions, we probably should pin to the current version.